showcase-app/main.js

```diff
@@ -54,30 +59,36 @@ const main = async () => {
   // You can get a list of all identities currently available from the chrome extension:
   // const identities = await identityWidget.getIdentities();
 
-  updateState();
+  await updateState();
 };
 
+// Invalidates the application - purely checks whether logged in or not and updates app state based on that
 const updateState = async () => {
   if (getDefaultSession().info.isLoggedIn) {
-    console.log('%cLOGGED IN', 'padding: 5px; border-radius: 3px; background: #e3c; font-weight: bold; color: white', getDefaultSession().info);
+    console.log(
+      '%cLOGGED IN',
+      'padding: 5px; border-radius: 3px; background: #e3c; font-weight: bold; color: white',
+      getDefaultSession().info,
+    );
     document.getElementById('app-card').classList.remove('hidden');
     document.getElementById('login-card').classList.add('hidden');
 
     const session = getDefaultSession();
 
-    // For example, the user must be someone with Read access to the specified URL.
-    const myDataset = await getSolidDataset(
-      session.info.webId,
-      { fetch }
-    );
+    // Getting dataset from a specific WebID (the one that was logged in with)
+    const myDataset = await getSolidDataset(session.info.webId, { fetch });
 
+    // Getting contents at /me from the current pod
     const me = getThing(myDataset, session.info.webId);
+
+    // Getting the name from the vcard if exists
     const name = getStringNoLocale(me, SCHEMA_INRUPT.name);
 
+    // By means of an example, we can pump back metadata from the pod to the extension using metadata and an update event
```

Comment:
I would not use expression like "pump back" in documentation, as not everybody might be familiar with them.